lib/reverse_sentence.rb

```diff
@@ -1,6 +1,37 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n) - where n is the length of input string
+# Space complexity: constant, O(1) - the space used remains the same regardless
+#                   of the input size
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+
+  start_index = 0
+  end_index = 0
+  length = my_sentence.length
+
+  while end_index < length
+    if my_sentence[end_index] == " " || end_index == length - 1
+      temp_index = end_index
```

Comment:
Personal-preference^ nit: move these into one line and use a ternary operator? e.g.
```
temp_index = end_index - (my_sentence[end_index] == " " ? 1 : 0)
```

You can also do something like this, but it's a _slight_ bit more complex:
```
temp_index = end_index - ((1 if my_sentence[end_index] == " ") or 0)
```

^ The usual disclaimer: some people like explicit code while others prefer concise code. There are pros and cons to both, so use whichever one suits you best. 🙂(That being said, it _is_ important to be able to _understand_ the approaches you're not used to using - and sometimes, the best practice is to try using them yourself! 😄)